### PR TITLE
NAS-127205 / 24.10 / Refactor SmartTestResultList

### PR DESCRIPTION
**Pull request by denysbutenko**

### CODE REVIEW

File: src/app/pages/storage/modules/disks/components/smart-test-result-list/smart-test-result-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+  ngOnInit(): void {
+    this.createDataProvider();
+  }
+
+  createDataProvider(): void {
+    const smartTestResults$ = this.ws.call('disk.query', [[], { extra: { pools: true } }]).pipe(
+      switchMap((disks) => {
+        this.disks = disks;
+        this.updateQueryParams();
```
Comment (undsoft) [inline src/app/pages/storage/modules/disks/components/smart-test-result-list/smart-test-result-list.component.ts:105]: Doesn't look like `queryParams` needs to be class property.

File: src/app/pages/storage/modules/disks/components/smart-test-result-list/smart-test-result-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+      map((smartTestResults: SmartTestResults[]) => {
+        const rows: SmartTestResultsRow[] = [];
+        smartTestResults.forEach((smartTestResult) => {
+          smartTestResult?.tests.forEach((test) => {
+            rows.push({ ...test, disk: smartTestResult.disk });
+          });
+        });
+        return rows;
+      }),
+      tap((smartTestResults) => this.smartTestResults = smartTestResults),
```
Comment (undsoft) [inline src/app/pages/storage/modules/disks/components/smart-test-result-list/smart-test-result-list.component.ts:117]: I don't suppose we can rely on `setFilter` in the data provider?

**Review by undsoft — APPROVED** [review]
Looks okay, though there is an empty details section on my screen:

Снимок экрана 2024-02-07 в 17 46 49

File: src/app/pages/storage/modules/disks/components/smart-test-result-list/smart-test-result-list.component.ts (hunk 2 above)
Comment (denysbutenko) [reply src/app/pages/storage/modules/disks/components/smart-test-result-list/smart-test-result-list.component.ts:117]: Never seen this method before. I'll note this to [Standardize common patterns in ix-table2](https://ixsystems.atlassian.net/browse/NAS-127168)

**Pull request merged**